uoftscrapers/scrapers/calendar/utm.py

```diff
@@ -4,14 +4,75 @@
 import json
 import os
 import requests
+import datetime
+now = datetime.datetime.now()
 
 
 class UTMCalendar:
+    '''Scraper for Important dates from UTM calendar found at https://www.utm.utoronto.ca/registrar/important-dates
+        '''
 
-    host = 'http://www.artsandscience.utoronto.ca/ofr/calendar/'
-
+    summerLink = 'http://m.utm.utoronto.ca/importantDates.php?mode=full&session={0}5&header='
+    fallLink = 'http://m.utm.utoronto.ca/importantDates.php?mode=full&session={0}9&header='
+    sessionLinks = [summerLink, fallLink]
+    currentSession = "Summer"
     @staticmethod
-    def scrape(location='.'):
+    def scrape(location='.', year=None): #scrapes most current sessions by default
+        
+        year = year or now.year
+
+        calendar = OrderedDict()
         Scraper.logger.info('UTMCalendar initialized.')
-        Scraper.logger.info('Not implemented.')
+        for link in UTMCalendar.sessionLinks:
+            html = Scraper.get(link.format(year))
+            soup = BeautifulSoup(html, 'html.parser')
+            content = soup.find('div', class_='content')
+            dates = content.find_all('div', class_='title')
+            i = 0
+            currentDate = dates[i]
+            while(i<len(dates)):
+                date = dates[i].text
+                events = []
+                while (currentDate == dates[i]):
+                    info = dates[i].find_next('div', class_='info')
+                    description = info.text
+                    eventStartEnd = date.split('-') #splits event dates over a period
+                    eventStart = eventStartEnd[0].strip()
+                    if len(eventStartEnd)>1:
+                        eventEnd = eventStartEnd[1].strip()
+                    else:
+                        eventEnd = eventStart
+
+                    events.append(OrderedDict([
+                            ('end_date', eventEnd),
+                            ('campus', 'UTM'),
+                            ('description', description)
+                        ]))
+                    i+=1
+                    if(i>=len(dates)):
```

Comment:
it actually checks for the case when we reach the final date in the array. It's kinda "hacky" way to solve the problem of the list going out of range. 

I originally used a for loop but I thought a while loop was easier to implement and better in this case. Please let me know if I should change it.